### DISCUSSION TITLE
Inserting data while performing a query, but querying old data will also slow the query

### DISCUSSION THREAD

**Discussion by 1292253144** [milvus-io/milvus#27819] [Q&A and General discussion]
Does simultaneously inserting data and searching impact query performance?
Insert operations are not CPU intensive. However, because new segments may not have reached the threshold for index building, Milvus resorts to brute-force search—significantly impacting query performance.

Inserting data while performing a query, but querying old data will also slow the query？

**Comment by yhmo** [comment]
"querying old data" -------- do you mean filtering the data with a timestamp?

Milvus manages data by segments, we have growing segments, and sealed segments. Inserted data is accumulated into growing segments. If a growing segment size reaches a threshold, it will be turned into a sealed segment.

- Milvus does brute-force for growing segments. (maybe in future versions support smart index for growing segments)
- If a sealed segment is not yet indexed, milvus will do a brute-force search for it. 

If your search expression can filter out all the growing segments, inserting data will not affect the search performance.
